manifests/profile/www_lib/vhosts/staff_lib.pp

```diff
@@ -0,0 +1,168 @@
+# Copyright (c) 2019 The Regents of the University of Michigan.
+# All Rights Reserved. Licensed according to the terms of the Revised
+# BSD License. See LICENSE.txt for details.
+
+# nebula::profile::www_lib::vhosts::staff_lib
+#
+# staff.lib.umich.edu virtual host
+#
+# @example
+#   include nebula::profile::www_lib::vhosts::staff_lib
+#
+class nebula::profile::www_lib::vhosts::staff_lib (
+  String $prefix,
+  String $domain,
+  String $ssl_cn = 'staff.lib.umich.edu',
+  # TODO: flatten
+  String $vhost_root = '/www/staff.lib',
+  String $docroot = "${vhost_root}/web"
+) {
+
+  # These directories are all served by PHP5 using mod_php and require Cosign
+  $php5_dirs = [
+    'coral',
+    'ptf',
+    'sites/staff.lib.umich.edu/local',
+  ].map |$dir| {
+    {
+      provider => 'directory',
+      path => "${docroot}/${dir}",
+      addhandlers => [{
+        extensions => ['.php'],
+        handler => 'application/x-httpd-php'
+      }],
+      custom_fragment => 'CosignAllowPublicAccess off'
+    }
+  }
+
+  nebula::apache::www_lib_vhost { 'staff.lib http redirect':
+    servername      => "${prefix}staff.${domain}",
+    ssl             => false,
+    docroot         => $docroot,
+    redirect_status => 'permanent',
+    redirect_source => '/',
+    redirect_dest   => 'https://staff.lib.umich.edu/'
+  }
+
+  nebula::apache::www_lib_vhost { 'staff.lib ssl':
+    servername                    => "${prefix}staff.${domain}",
+    ssl                           => true,
+    usertrack                     => true,
+    cosign                        => true,
+    docroot                       => $docroot,
+
+    aliases                       => [
+      { scriptalias => '/cgi', path => "${vhost_root}/cgi" }
+    ],
+
+    directories                   => [
+      {
+        provider       => 'directory',
+        path           => $docroot,
+        options        => ['IncludesNOEXEC','Indexes','FollowSymLinks','MultiViews'],
+        allow_override => ['AuthConfig','FileInfo','Limit','Options'],
+        require        => $nebula::profile::www_lib::apache::default_access,
+        addhandlers    => [{
+          extensions => ['.php'],
+          # TODO: Extract version or socket path to params/hiera
+          handler    => 'proxy:unix:/run/php/php7.3-fpm.sock|fcgi://localhost'
+        }],
+      },
+      {
+        provider       => 'directory',
+        path           => "${vhost_root}/cgi",
+        allow_override => ['None'],
+        options        => ['None'],
+        require        => $nebula::profile::www_lib::apache::default_access,
+      },
+      {
+        provider       => 'directory',
+        path           => "${vhost_root}/alida-tmp/current/public",
+        allow_override => ['None'],
+        options        => ['FollowSymlinks'],
+        require        => $nebula::profile::www_lib::apache::default_access,
+      },
+      {
+        # Deny access to raw php sources by default
```

Comment:
This might be already configured via a top-level apache include, not sure...

Reply:
Thanks for flagging this. It doesn't appear to be anywhere more general, and I'm not ready to rely on the module for invisible defaults, so I'll leave this here.